### PR TITLE
Fix no implicit any errors (Part 3 of 8)

### LINKED ISSUE #5620: Add noImplicitAny to server tsconfig (Part 3 of 8)
When we migrated to typescript from javascript, it was too expensive to turn on `noImplicitAny` because it was hard enough just making the switch.

Today, having `noImplicitAny` could catch a lot of errors that I'm seeing in code reviews.
The problem is that there are 1600 errors!
This is too many for any one person to do in a sprint.
However, the benefit of having this turned on for new code is pretty great.

So, I propose we share the love & chip away at it:
- [ ] turn on `noImplicitAny` in the server tsconfig file
- [ ] fix ~200 errors
- [ ] turn it off again in the tsconfig file & submit the PR
- [ ] we'll merge it blindly
- [ ] create a copy of this issue named Part (3 of 8) so we can re-prioritize for the next person. We'll keep doing this until it's done

AC
- [ ] 200+ `noImplicitAny` errors are fixed in the server package

Estimate: 4 hours

### PR DESCRIPTION
**Pull request by nickoferrall**
Fix #5620 

Following [this comment](https://github.com/ParabolInc/parabol/issues/5314#issuecomment-923128109), I'll move this PR straight to Maintainer Review for a light CR. 

I've created an issue for part 4 here: https://github.com/ParabolInc/parabol/issues/5696

### CODE REVIEW

File: packages/server/graphql/types/PokerTemplate.ts [hunk 1]
```diff
@@ -17,9 +17,9 @@ const PokerTemplate = new GraphQLObjectType<any, GQLContext>({
     dimensions: {
       type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(TemplateDimension))),
       description: 'The dimensions that are part of this template',
-      resolve: async ({id: templateId}, _args, {dataLoader}) => {
+      resolve: async ({id: templateId}, _args: unknown, {dataLoader}) => {
         const dimensions = await dataLoader.get('templateDimensionsByTemplateId').load(templateId)
-        const activeDimensions = dimensions.filter(({removedAt}) => !removedAt)
+        const activeDimensions = dimensions.filter(({removedAt}: {removedAt: Date}) => !removedAt)
```
Comment (mattkrick) [inline packages/server/graphql/types/PokerTemplate.ts:22]: this one seems odd, should already be `Date | undefined`. were you getting a noImplicityAny here too?

File: packages/server/graphql/types/RetrospectiveMeetingSettings.ts [hunk 2]
```diff
@@ -69,7 +73,7 @@ const RetrospectiveMeetingSettings = new GraphQLObjectType<any, GQLContext>({
         const {orgId} = team
         const templates = await dataLoader.get('meetingTemplatesByOrgId').load(orgId)
         const organizationTemplates = templates.filter(
-          (template) =>
+          (template: MeetingTemplate) =>
```
Comment (mattkrick) [inline packages/server/graphql/types/RetrospectiveMeetingSettings.ts:76]: this is another odd one! it should already be a MeetingTemplate without the cast, right?

File: packages/server/graphql/types/SetPhaseFocusPayload.ts [hunk 3]
```diff
@@ -18,9 +19,9 @@ const SetPhaseFocusPayload = new GraphQLObjectType<any, GQLContext>({
     },
     reflectPhase: {
       type: new GraphQLNonNull(ReflectPhase),
-      resolve: async ({meetingId}, _args, {dataLoader}) => {
+      resolve: async ({meetingId}, _args: unknown, {dataLoader}) => {
         const meeting = await dataLoader.get('newMeetings').load(meetingId)
```
Comment (mattkrick) [inline packages/server/graphql/types/SetPhaseFocusPayload.ts:23]: nit: 

```ts
const meeting = (await dataLoader
          .get('newMeetings')
          .load(meetingId)) as MeetingRetrospective
```

will make the error down below go away (not sure why type unions give this error in typescript anyways).

File: packages/server/graphql/types/Threadable.ts [hunk 4]
```diff
@@ -35,21 +35,22 @@ export const threadableFields = () => ({
   createdByUser: {
     type: require('./User').default,
     description: 'The user that created the item',
-    resolve: ({createdBy}, _args, {dataLoader}: GQLContext) => {
+    resolve: ({createdBy}: {createdBy: Date}, _args: unknown, {dataLoader}: GQLContext) => {
```
Comment (mattkrick) [inline packages/server/graphql/types/Threadable.ts:38]: this is a string

File: packages/server/graphql/types/UpdateUserProfilePayload.ts [hunk 5]
```diff
@@ -19,9 +19,9 @@ const UpdateUserProfilePayload = new GraphQLObjectType<any, GQLContext>({
     teamMembers: {
       type: new GraphQLList(new GraphQLNonNull(TeamMember)),
       description: 'The updated team member',
-      resolve: ({teamIds, userId}, _args, {dataLoader}) => {
+      resolve: ({teamIds, userId}, _args: unknown, {dataLoader}) => {
```
Comment (mattkrick) [inline packages/server/graphql/types/UpdateUserProfilePayload.ts:22]: nit: always nice to declare the type as early as possible, that way if we do something else with teamIds then we don't have to do multiple casts down below

**Review by mattkrick — CHANGES_REQUESTED** [review]
just 1 bug, the rest are nits. thanks so much for this!

File: packages/server/graphql/types/PokerTemplate.ts (hunk 1 above)
Comment (nickoferrall) [reply packages/server/graphql/types/PokerTemplate.ts]: I was! It knows the type but still shouts at me 🤷‍♂️  I've changed `Date` to `Date | undefined`

removed-at

File: packages/server/graphql/types/RetrospectiveMeetingSettings.ts (hunk 2 above)
Comment (nickoferrall) [reply packages/server/graphql/types/RetrospectiveMeetingSettings.ts:76]: Same as above - it knows the type but still complains!

**Pull request merged**